### ISSUE TITLE
Feature request: Run jobs independently

### ISSUE THREAD

**Issue by rkarlsba** [duplicati/duplicati#2775]
Hi. I've recently started using Duplicati, and what's rather annoying is that the jobs aren't run in parallel, but serielised. This is particularly bad if you try to remove a job while another is running. Also, parallel jobs may speed up uploads to target (given you have sufficient of resources). I've read about the parallel version, but it seem it still doen't run jobs in parallel. I beleive the project could benefit from such a change.

**Comment by kenkendk** [comment]
It would require some encapsulation to get it working. For instance, the SSL certificate overrides are implemented in a way that is only possible "pr process".

It is possible to use AppDomains to encapsulate each backup, and that should solve all the issues, but this requires that the status reporting is rewritten to support this.

**Comment by rkarlsba** [comment]
Would it be easier to allow for parallel uploads to different targets? The current code seems extemely serielaised, which isn't exactly 2017.

**Comment by rkarlsba** [comment]
Any updates on this? It would quite possibly speed up backups dramatically.

**Comment by restinplace** [comment]
Add another vote for this (going to Amazon Drive, fwiw).
And thanks for a great tool!

**Comment by archon810** [comment]
Another vote for parallel jobs so you could do a restore while doing a backup.

Heck, even viewing a log fails because of a locked database.

**Comment by lairdb** [comment]
Tangent suggestion: what about an alternative or interim step of job push-in?  IOW, if a job with priority 3 was running, and a job with priority 2 wanted to start, the running job would "stop after current file" and re-queue itself, allowing job 2 to run and passively continuing job 3 when job 2 was done.

(I can write this up as its own ticket if feedback is positive.)
